R/corridor.R

```diff
@@ -97,40 +104,72 @@ river_buffer <- function(river, buffer, bbox = NULL) {
   river_buf <- sf::st_buffer(river, buffer)
   river_buf_union <- sf::st_union(river_buf)
   if (!is.null(bbox)) {
-    return(sf::st_crop(river_buf_union, bbox))
+    sf::st_crop(river_buf_union, bbox)
   } else {
-    return(river_buf_union)
+    river_buf_union
   }
 }
 
-#' Draw a corridor as an estimate of the river valley.
-#'
-#' The valley is drawn on a digital elevation model (DEM) of the area,
-#' see [get_valley()] for details on the implementation.
-#'
-#' @param river A simple feature geometry representing the river
-#' @param dem The DEM of the area as a terra SpatRaster object
-#' @param bbox Bounding box defining the extent of the area of interest
-#'
-#' @return A simple feature geometry
-river_valley <- function(river, dem, bbox = NULL) {
-  get_valley(dem, river, bbox = bbox)
-}
-
 #' Find the corridor end points.
 #'
-#' Using the river center line and the bounding box of the area of interest,
-#' determine the extremes (start and end point) of the river corridor.
+#' Determine the extremes (start and end point) of the river corridor
+#' using the river center line and the spatial network used for the delineation.
+#' The end points are selected as the most external river crossing edges of the
+#' network within the area of interest. Note that if the river intersects the
+#' area of interest multiple times, only the longest intersecting segment is
+#' considered here.
 #'
 #' @param river A simple feature geometry representing the river centerline
-#' @param bbox Bounding box defining the extent of the area of interest
+#' @param network The spatial network used for the delineation, either provided
+#'   as a [`sfnetworks::sfnetwork`] or as a [`sf::sf`]/[`sf::sfc`] object (edges
+#'   only)
+#' @param aoi Area of interest, provided as a bounding box or as polygon. If not
+#'   given, it is taken as the bounding box of the spatial network
 #'
 #' @return A simple feature geometry including a pair of points
-corridor_end_points <- function(river, bbox) {
-  aoi <- sf::st_as_sfc(bbox)
-  aoi_boundary <- sf::st_boundary(aoi)
-  intersections <- sf::st_intersection(river, aoi_boundary)
-  sf::st_cast(intersections, "POINT")
+corridor_end_points <- function(river, network, aoi = NULL) {
+  # Only consider the edges of the spatial network
+  spat_net_edges <- sf::st_geometry(sf::st_as_sf(network, "edges"))
+
+  # Set area of interest as a polygon, even if not defined
+  if (is.null(aoi)) aoi <- sf::st_bbox(spat_net_edges)
+  aoi <- as_sfc(aoi)
+
+  # Clip the river geometry using the area of interest. This might lead to
+  # multiple river segments
+  river_segments <- sf::st_intersection(aoi, sf::st_cast(river, "LINESTRING"))
```

Comment:
I see that you use nested functions instead of dplyr-style piped expressions. This is more compact but can be less readable with multiple function calls. We use the pipe in some other places (see, e.g., the valley module). I have a slight preference for the latter as it is more readable, even with two function calls. What do you think?

Or is the choice related to performance as well? 

This is not something to solve now (it may actually be a nice first issue for someone new in the team on my side). We can create an issue on consistency in nesting/piping multiple functions and we can add that in the contributor guidelines.